browser/brave_wallet/brave_wallet_tab_helper.cc

```diff
@@ -34,6 +62,34 @@ void BraveWalletTabHelper::ShowBubble() {
   wallet_bubble_manager_delegate_->ShowBubble();
 }
 
+void BraveWalletTabHelper::UserRequestCompleted(size_t hash,
+                                                const std::string& result) {
```

Comment:
Will this always be called for each request? What happens if user switch tabs without approve/cancel in the bubble?

Reply:
And we need tests to ensure our expected behavior is true.